gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupRequestHandler.java

```diff
@@ -109,37 +109,33 @@ private String collectFailureReason(final List<PartitionBackupStatus> partitionS
         .collect(Collectors.joining());
   }
 
-  private BackupStatusCode getAggregatedStatus(
-      final List<PartitionBackupStatus> partitionStatuses) {
-    return partitionStatuses.stream()
-        .map(PartitionBackupStatus::status)
-        .reduce(this::combine)
-        .orElseThrow(
-            () ->
-                // This should never happen, because partitionStatuses would never be empty.
-                new IllegalStateException(
-                    "Backup status cannot be calculated from status of partitions backup %s. Possible incomplete topology."
-                        .formatted(partitionStatuses)));
-  }
-
-  private BackupStatusCode combine(final BackupStatusCode x, final BackupStatusCode y) {
-    // Failed > DoesNotExist > InProgress > Completed
+  private State getAggregatedStatus(final List<PartitionBackupStatus> partitionStatuses) {
+    final var statuses =
+        partitionStatuses.stream().map(PartitionBackupStatus::status).distinct().toList();
 
-    if (x == BackupStatusCode.FAILED || y == BackupStatusCode.FAILED) {
-      return BackupStatusCode.FAILED;
+    if (statuses.contains(BackupStatusCode.FAILED)) {
+      return State.FAILED;
     }
-    if (x == BackupStatusCode.DOES_NOT_EXIST || y == BackupStatusCode.DOES_NOT_EXIST) {
-      return BackupStatusCode.DOES_NOT_EXIST;
+    if ((statuses.contains(BackupStatusCode.IN_PROGRESS)
+            || statuses.contains(BackupStatusCode.COMPLETED))
+        && statuses.contains(BackupStatusCode.DOES_NOT_EXIST)) {
+      return State.INCOMPLETE;
     }
-    if (x == BackupStatusCode.IN_PROGRESS || y == BackupStatusCode.IN_PROGRESS) {
-      return BackupStatusCode.IN_PROGRESS;
+    if (statuses.contains(BackupStatusCode.IN_PROGRESS)) {
+      return State.IN_PROGRESS;
     }
-    if (x == BackupStatusCode.COMPLETED && y == BackupStatusCode.COMPLETED) {
-      return BackupStatusCode.COMPLETED;
+    if (statuses.contains(BackupStatusCode.COMPLETED)) {
+      return State.COMPLETED;
+    }
+
+    if (statuses.contains(BackupStatusCode.DOES_NOT_EXIST)) {
```

Comment:
:wrench: I think it's currently not needed but for safety reasons I'd suggest flipping the order of the checks for `COMPLETED` and `DOES_NOT_EXIST`. If `statuses.contains(BackupStatusCode.DOES_NOT_EXIST)` we always want to return `DOES_NOT_EXIST` (or `INCOMPLETE` but that's already covered).

Alternatively, only return `COMPLETED` if all statuses are `COMPLETED`

Reply:
:+1: